src/core/public/apm_system.ts

```diff
@@ -87,6 +74,45 @@ export class ApmSystem {
     });
   }
 
+  /* Hold the page load transaction open, until all resources actually finish loading */
+  private holdPageLoadTransaction(apm: ApmBase) {
+    this.pageLoadTransaction = apm.getCurrentTransaction();
```

Comment:
Maybe you should store the transaction after `this.pageLoadTransaction.type === 'page-load'` check to avoid the check duplication in `closePageLoadTransaction` and `markPageLoadStart`?